### PR TITLE
feat(ubiquiti): add eth interfaces template

### PR DESCRIPTION
**Pull request by guillaume-mbali**
**ISSUE TYPE**
New Template Pull Request

**COMPONENT**
ubiquiti_edgerouter_show_interfaces.textfsm, edgerouter, show interfaces

**SUMMARY**
Added the ubiquiti edgerouter template to show eth interfaces

### CODE REVIEW

File: ntc_templates/templates/ubiquiti_edgerouter_show_interfaces.textfsm [hunk 1]
```diff
@@ -0,0 +1,25 @@
+Value Filldown INTERFACE (\w+)
+Value DESCRIPTION ([\w\d\-]+)
```
Comment (elavaud) [inline ntc_templates/templates/ubiquiti_edgerouter_show_interfaces.textfsm]: Put DESCRIPTION under LINK_STATUS to maintain order

File: ntc_templates/templates/ubiquiti_edgerouter_show_interfaces.textfsm [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  ^\s+${IPV4_ADDRESS} -> Continue
+  ^\s+${IPV6_ADDRESS} -> Record
+  ^${INTERFACE}\s+${IPV4_ADDRESS}\s+(${STATE}/${LINK_STATUS}) -> Continue
+  ^${INTERFACE}\s+${IPV6_ADDRESS}\s+(${STATE}/${LINK_STATUS}) -> Record
+  ^\s+${IPV4_ADDRESS} -> Continue
+  ^\s+${IPV6_ADDRESS} -> Record
+  ^${INTERFACE}\s+${IPV4_ADDRESS}\s+(${STATE}/${LINK_STATUS}) -> Continue
+  ^${INTERFACE}\s+${IPV6_ADDRESS}\s+(${STATE}/${LINK_STATUS}) -> Record
+  ^\s+${IPV4_ADDRESS} -> Continue
+  ^\s+${IPV6_ADDRESS} -> Record
```
Comment (elavaud) [inline ntc_templates/templates/ubiquiti_edgerouter_show_interfaces.textfsm]: There are way too much possibilities.
You should be able to do something much cleaner with "Continute.record" like in huawei_vrp_display_acl_all
You can use also "Required" to indicate the value need to be collected before recording.

Those 15 lines can be reduced to 3 or 5

You can also check huawei_vrp_display_interface_description for that thing of Starting the recording only after a certain line. You would ensure you would start recording right after
"Interface    IP Address                        S/L  Description "
and you would be able to include the first lines

File: tests/ubiquiti_edgerouter/show_interfaces/ubiquiti_edgerouter_show_interfaces.raw [hunk 3]
```diff
@@ -0,0 +1,10 @@
+Codes: S - State, L - Link, u - Up, D - Down, A - Admin Down
+Interface    IP Address                        S/L  Description                 
+---------    ----------                        ---  -----------                 
+eth0         -                                 u/u  Port                        
+eth1         192.168.1.1/24                    u/u  datas                                
+             2a05:c100:1d::1/64
+eth2         -                                 u/D  Port                        
+eth3         -                                 u/D  Port
+lo           127.0.0.1/8                       u/u                              
+             ::1/128
```
Comment (elavaud) [inline tests/ubiquiti_edgerouter/show_interfaces/ubiquiti_edgerouter_show_interfaces.raw]: Add 2 dummy interfaces, one with only an IPv4 address, and one with only an IPv6 address.
You could use eth2 and eth3. In that way you would have all cases.

Also, make one Administratively down

**Review by elavaud — CHANGES_REQUESTED** [review]
Rename commit. It is for NTC template, not us. Like:
feat(ubiquiti): Add template ubiquiti edgerouter show interfaces

**Review by elavaud — APPROVED** [review]
Ok for me.
You can convert to draft, put it on NTC side, add the tags on this PR, and cherry-pick the commit in our v3.1.0 branch

**Comment by guillaume-mbali** [comment]
Official release : [#1291](https://github.com/networktocode/ntc-templates/pull/1291)

**Pull request closed without merging**